tests/Unit/API/Google/MerchantTest.php

```diff
@@ -0,0 +1,321 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\API\Google\Merchant;
+use Automattic\WooCommerce\GoogleListingsAndAds\Exception\MerchantApiException;
+use Automattic\WooCommerce\GoogleListingsAndAds\Options\OptionsInterface;
+use Automattic\WooCommerce\GoogleListingsAndAds\Tests\Framework\UnitTest;
+use Exception;
+use Google\Exception as GoogleException;
+use Google\Service\ShoppingContent;
+use Google\Service\ShoppingContent\Account;
+use Google\Service\ShoppingContent\AccountAdsLink;
+use Google\Service\ShoppingContent\AccountStatus;
+use Google\Service\ShoppingContent\AccountUser;
+use Google\Service\ShoppingContent\Product;
+use Google\Service\ShoppingContent\ProductsListResponse;
+use Google\Service\ShoppingContent\ProductstatusesCustomBatchResponse;
+use Google\Service\ShoppingContent\Resource\Accounts;
+use Google\Service\ShoppingContent\Resource\Accountstatuses;
+use Google\Service\ShoppingContent\Resource\Products;
+use Google\Service\ShoppingContent\Resource\Productstatuses;
+use PHPUnit\Framework\MockObject\MockObject;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class MerchantTest
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tests\Unit\API\Google
+ *
+ * @property  MockObject|ShoppingContent  $service
+ * @property  MockObject|OptionsInterface $options
+ * @property  Merchant                    $merchant
+ */
+class MerchantTest extends UnitTest {
+
+	/**
+	 * Runs before each test is executed.
+	 */
+	public function setUp() {
+		parent::setUp();
+		$this->service           = $this->createMock( ShoppingContent::class );
+		$this->service->accounts = $this->createMock( Accounts::class );
+		$this->service->products = $this->createMock( Products::class );
+
+		$this->options  = $this->createMock( OptionsInterface::class );
+		$this->merchant = new Merchant( $this->service );
+		$this->merchant->set_options_object( $this->options );
+	}
+
+	public function test_get_products_empty_list() {
+		$list_response = $this->createMock( ProductsListResponse::class );
+
+		$this->service->products->expects( $this->any() )
+			->method( 'listProducts' )
+			->willReturn( $list_response );
```

Comment:
Personally I'd be considering using the `->with()` method to explicitly set the arguments for a few of the method expectations here. 

```suggestion
		$merchant_id = uniqid();
		$this->options->method( 'get_merchant_id' )->willReturn( $merchant_id );
		
		$this->service->products->expects( $this->any() )
			->method( 'listProducts' )
			->with( $merchant_id )
			->willReturn( $list_response );
			
			
```